src/profile-logic/merge-compare.js

```diff
@@ -132,21 +132,42 @@ export function mergeProfilesForDiffing(
       );
     }
     const profile = profiles[i];
-    let thread = profile.threads[selectedThreadIndex];
+    let thread = { ...profile.threads[selectedThreadIndex] };
     transformStacks[i] = profileSpecific.transforms[selectedThreadIndex];
     implementationFilters.push(profileSpecific.implementation);
 
     // We adjust the categories using the maps computed above.
     // TODO issue #2151: Also adjust subcategories.
-    thread.stackTable.category = adjustCategories(
-      thread.stackTable.category,
-      translationMapsForCategories[i]
-    );
-    thread.frameTable.category = adjustNullableCategories(
-      thread.frameTable.category,
-      translationMapsForCategories[i]
+    thread.stackTable = {
+      ...thread.stackTable,
+      category: adjustCategories(
+        thread.stackTable.category,
+        translationMapsForCategories[i]
+      ),
+    };
+    thread.frameTable = {
+      ...thread.frameTable,
+      category: adjustNullableCategories(
+        thread.frameTable.category,
+        translationMapsForCategories[i]
+      ),
+    };
+
+    const screenshotMarker = profile.threads.filter(
+      t => t.name === 'Compositor'
+    )[0].markers;
+
+    const compositorScreenshot = screenshotMarker.data.filter(
+      d => d !== null && d.type === 'CompositorScreenshot'
     );
 
+    /*
+    const mergeScreenshot = { ...thread.markers, ...screenshotMarker };
+    thread.markers = mergeScreenshot;
+    */
+
+    Array.prototype.push.apply(thread.markers.data, screenshotMarker.data);
```

Comment:
This merges the data, I can access those from the console. After trying this, I noticed that the data array of the `thread.marker` changes, which is obvious, so I was thinking if I need to do this for other property arrays() of the markers object too? 
*These are all my noob thoughts, sorry if this doesn't make sense 😅

Reply:
So right now it looks like we are pushing the `data` of screenshot markers to `data` of the thread we would like to send. But here, we would also push the `name`, `startTime`, `endTime`, `phase`, and `category`. That will pretty much going to look like mergeMarkers in the end actually. I think you can reuse some of its code on your own function.

So I think we can to create a function called `mergeScreenshotMarkers`. That will be pretty similar for `mergeMarkers` but it will be more specific to screenshots. We can't use that directly because apparently it will not for that.

I think I mentioned before, `url` field of the screenshot string is not the url string, it's only an index into the stringTable. So we need to get that string from old stringTable and copy it into the new one. To do that, you can take a look at the current code for marker "name"s. They are also index into string tables, and we need to copy them as well like this:
https://github.com/firefox-devtools/profiler/blob/319253c5757a85737f813866e63540c7426f4be5/src/profile-logic/merge-compare.js#L1078-L1079
getString adds the string if it's not present in the stringTable already.
And here we get the new stringTable index:
https://github.com/firefox-devtools/profiler/blob/319253c5757a85737f813866e63540c7426f4be5/src/profile-logic/merge-compare.js#L1084
So we need to do the exact same thing, but for `data.url` of screenshot markers.

Hope that helps! Let me know if you have any questions!